### DISCUSSION TITLE
Preserve valuable DTag from being registered

### DISCUSSION THREAD

**Discussion by rmontagnin** [desmos-labs/desmos#561] [Features ideas]
Since Desmos will allow any user to register their DTag from day one, this might cause some users to abuse this freedom and register multiple valuable DTags (eg. short ones or the ones associated with real VIPs). This has already happened in centralized social networks were very short usernames are now worth a ton of money. 

I think we should define some ways to prevent this, to make sure that users cannot register those DTags only to re-sell them in the future. Me and @g-luca have thought that in order to solve we could:

1. Force the length of the registered DTag to be at least 6 characters long.  
   This way we can preserve short ones and maybe sell them later with some kind of auction. 
2. Set a list of DTags that cannot be used.  
   This should help us making sure that none uses tags like `elonmusk`, `rogerfederer` etc. This list can be present inside the `x/profiles` parameters and updated through governance proposals. 

To have the best prevention against this kind of economic attack, we should implement those methods together so that all tags with less than 6 characters + all VIP tags can be preserved and sold later. 

**Note**  
In the future, the DTag auctions can be used to gather money for different social events (eg. like what Starname did selling short names to raise funds against the COVID pandemic). 

Please let me know what you think @kwunyeung @bragaz @dadamu

**Comment by leobragaz** [comment]
This make a lot of sense, considering that a person can have multiple desmos addresses and profiles possibly collecting a lot of valuable DTags. It's worth to explore DTag auctions, maybe managed by a smart contract?
Btw, if we force DTags to be at least 6 chars long to preserve those short ones, we need to rethink a little the `profiles` params, cause there could be some problems if for example we have the length param set to 6 and then we got some DTags with 3 chars. This can easily break the invariants if we don't pay attention.

**Comment by kwunyeung** [comment]
Oh haha! I suggested this when I first mentioned the idea of `DTag` but you guys thought it would affect the free market and convinced me not to have this feature. The example I brought out was Handshake.org. They reserved [100,000 domain names](https://github.com/handshake-org/hs-names#handshake-names) form the top 1m visited websites on Alexa. They will wait until the owners of the original domain names claim those names. If you search those domain names, e.g. apple, on Handshake, you will see this.

![image](https://user-images.githubusercontent.com/932636/126315208-ddfcb090-92ca-4bde-9efa-626d0c4d4c9a.png)

They even don't have an estimated release date of the domain name. It won't be available for bidding. And this apply to `desmos/` on Handshake as well.

I do agree to have such arrangement. The problem is how we generate the list of preserved `DTag`. How to get popular usernames from `Twitter`, `Facebook`, `Instagram`, `Youtube`, `Reddit`?

For `DTag` auction, I do would like to see if this can be done with smart contract. If this will be managed by SC, then the transfer can only be available via SC but not original messages like now. Also, if we make it like auctions, we can add NFT elements in it. For every exchange, the original minter of the `DTag` could have a share of that exchange? We also mentioned about taking a portion of the exchange amount as community tax.

**Reply by rmontagnin** [reply]
I would like to get @bragaz and @dadamu to work on this question specifically: 

>  how we generate the list of preserved DTag. How to get popular usernames from Twitter, Facebook, Instagram, Youtube, Reddit?

Could you please try and see if there is any existing list of most popular names used? What I would like to have is a list of usernames used by VIPs

**Reply by leobragaz** [reply]
looking online I've found out that twitter has its own lists of verified accounts divided in categories:
https://twitter.com/verified/lists.
Unfortunately I haven't got the same luck with others SN. There's seems to be no other lists of verified accounts.
Talking about companies we can possibly get those from the S&P 500 and any other known indexes.
I will research more to find out if there's a feasible way to collect these data

**Reply by kwunyeung** [reply]
Hm… that list doesn’t make any sense to me. Seems it’s only a list that maintained by @verified which are not verified Twitter accounts.

**Reply by dadamu** [reply]
I have found the site gathering the most followed profiles rank here. I suppose we can refer it to build the list. 
https://www.trackalytics.com/

**Comment by dadamu** [comment]
From my side, Avoiding users registering famous or popular names in the anonymous network doesn't make sense to me. We might be able to set them to the list but it would include only a little part of them, and the users having the same name in the list must change another name since the list will reject them. The idea sounds that the list protects the famous people and the DTag economy system. However, the DTag troll still can register other popular names. Say, `Chen` is the popular last name in my country, the names in English are duplicated easily so the troll can register a set of profiles having DTags like `paulchen`, `davidchen`, `robertchen` and etc.

Honestly, human-readable name should not be unique in my opinion. In my experience, allowing duplicated name in the system is the trend. Many of games applies the way to assign the unique id and let users set their name. In this model, wallet address is the unique id already. Why don't we simply let DTag duplicated or set DTag has random number suffix like `paulchen#7777` as Discord does?

**Reply by rmontagnin** [reply]
To be honest, I have the same concerns as you. I'm internally very debated on whether we should adopt this style of preserving specific DTags or not, as it has its pros and cons (as everything). 

**PROs**
- Limits the number of trolls/fake accounts/impersonations

**CONs**
- Limits the freedom of users
- Might create a non-fair market for DTags

Initially, I proposed to not adapt this kind of limitations and simply have the users be free to register whatever DTag they want. However, I notices that someone is already using well-known names to impersonate VIPs in the testnet (there is an `elonmusk` registered DTag). For this reason, I wanted to see your opinions about this. 

This might not be a problem at all if we can safely rely on third-party authentication like Twitter etc. Those fake accounts won't be able to connect their profile to the Twitter account and thus will always be seen as "non verified" (and thus to be considered a scam). 

Also, in order to implement this limitation we would need to find the top X most followed accounts of people, companies, etc which might be extremely hard to do and might make the chain very slow during the process of creating/saving a profile.

In the end, I think the best way to avoid any problem of the DTag is to simply limit the length by not allowing people to use very short ones (i.e. those with 6 or less characters) and simply leave them for later auctions. For all the others, we simply rely on external verification. What do you think @bragaz @kwunyeung? 

---

Side note on Discord usernames.

Actually Discord does not allow any number of users having the same username. The part after the `#` is called a _discriminator_ and it is composed of only 4 digits. This means that at most 10000 people will be able to have the same username. If all of the slots are occupied, the 10001th user is going to be told that too many people are using the same username and that they must select another one. I don't think it would be of any value for Desmos to implement something like this since it might complicate the inner process during the profile creation/saving.

**Reply by dadamu** [reply]
I agree relying on third party platform is the better way to the issue. The trust should be built by the guarantor who stakes a lot of resource.  Initially, relying third party platform is the best selection for the case since they built for it with a lot of resource and the proof is already there.

For the Discord usernames sample, I consider that the troll would not appear if occupying a name is too costly. However, as you said, the inner process to deal with it is complicated.  And true, It might not be so much value if the profile creation/saving is not costly.

**Reply by leobragaz** [reply]
I agree with @kwunyeung when he said that the verified lists on twitter are useless for our purposes, plus there are no feasible ways to build a good list like we intended to. I think the best way to handle this would be by using the third-party authentication as a proof as you guys stated, since there's no way to trick that system. I would also avoid to add further complexity to the DTag system.

**Reply by kwunyeung** [reply]
I prefer we skip this limitation if we cannot find a good solution to generate the list of existing accounts. And I agree with @dadamu that this feature will become a protection to the existing famous people. I have rethink the same situation on Handshake.org. If they didn’t reserve `desmos/`, I have already registered it. Why they reserved it for the owner of `desmos.com` but not us who would sincerely appreciate the decentralized domain name? 

I registered `elonmusk` to demonstrate we don’t trust by only looking at a username. Also if there is a trend of registering popular `DTag` may become a good marketing tool as we will see a lot of `DTag` registrations in short time. It may help the future `DTag` exchange marketplace and benefits to the community as all trades should be taxed and contribute to the community pool. Active trading of `DTag` will be something we would be happy to see.

To protect some of our early supporters, how about we let teammates, validators and contributors to register their `DTag` on testnet and we migrate them on mainnet genesis?

**Reply by rmontagnin** [reply]
> To protect some of our early supporters, how about we let teammates, validators and contributors to register their DTag on testnet and we migrate them on mainnet genesis?

I don't think this would be fair either. Also, inside the testnet the DTag params are different from what will be in the mainnet. Inside the testnet you are allowed to register a DTag if that is long at least 3 characters, while in the mainnet we will be setting it to be at least 6. I think if everyone starts from the same base it would be more fair

**Reply by kwunyeung** [reply]
Okay. Let’s start mainnet with everyone the same.
